opm/material/common/FastSmallVector.hpp

```diff
@@ -137,10 +139,46 @@ class FastSmallVector
     const ValueType& operator[](size_t idx) const
     { return dataPtr_[idx]; }
 
-    //! number of the element
-    size_t size() const
+    using size_type = typename std::vector<ValueType>::size_type;
+
+    //! number of elements
+    size_type size() const
     { return size_; }
 
+    //! Iterator type is a plain pointer, so be warned there is no validity checking.
+    using Iterator = ValueType*;
+
+    //! To support range-for etc.
+    Iterator begin() const
+    { return dataPtr_; }
+
+    //! To support range-for etc.
+    Iterator end() const
+    { return dataPtr_ + size_; }
+
+    size_type capacity()
+    { return size_ < N ? N : data_.capacity(); }
```

Comment:
Might we want to have `size_ <= N` as the conditional here?  We don't really switch to the dynamic array until the size exceeds the static capacity.

Reply:
Good catch! I have also added a test that would have detected this bug.